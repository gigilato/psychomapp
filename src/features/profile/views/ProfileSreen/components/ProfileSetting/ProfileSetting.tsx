import clsx from 'clsx'

import { Box, Pressable, Icon, Text } from '$atoms'
import { tagColors } from '$theme'

import { IProfileSettingProps } from './ProfileSettings.props'

export const ProfileSetting = ({
  content,
  type = 'info',
  color = 'black',
  icon,
  className,
  ...props
}: IProfileSettingProps) => {
  return (
    <Pressable
      animation={type === 'info' ? 'none' : 'opacity'}
      className={clsx('flex-row items-center py-s', className)}
      {...props}
    >
      {icon ? (
        <Box
          className={clsx('bg-primary-light p-xs mr-s rounded-s')}
          style={{ backgroundColor: tagColors[color].light }}
        >
          <Icon
            name={icon}
            className={clsx('text-primary-strong')}
            style={{ color: tagColors[color].strong }}
            size={15}
          />
        </Box>
      ) : null}
      <Text variant="body" className="flex-1">
        {content}
      </Text>
      {type === 'info' ? null : (
        <Icon name={type === 'link' ? 'log-in' : 'chevron-right'} size={20} />
      )}
    </Pressable>
  )
}
