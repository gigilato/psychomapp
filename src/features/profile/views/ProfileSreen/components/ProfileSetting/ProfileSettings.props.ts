import { IPressableProps, IconName } from '$atoms'
import { TagColor } from '$theme'

export interface IProfileSettingProps extends IPressableProps {
  content: string
  onPress?: () => void
  icon?: IconName
  color?: TagColor
  type?: 'navigation' | 'link' | 'info'
}
