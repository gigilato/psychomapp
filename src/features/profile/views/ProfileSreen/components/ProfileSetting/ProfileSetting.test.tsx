import { render } from '@testing-library/react-native'
import React from 'react'

import { ProfileSetting } from './ProfileSetting'

describe('ProfileSetting', () => {
  it('should renders correctly', () => {
    const { toJSON } = render(<ProfileSetting ID="" content="content" />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should renders with icon', () => {
    const { toJSON } = render(<ProfileSetting ID="" content="content" icon="activity" />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should renders with link', () => {
    const { toJSON } = render(
      <ProfileSetting ID="" content="content" icon="activity" type="link" />
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should renders with navigation', () => {
    const { toJSON } = render(
      <ProfileSetting ID="" content="content" icon="activity" type="navigation" />
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
