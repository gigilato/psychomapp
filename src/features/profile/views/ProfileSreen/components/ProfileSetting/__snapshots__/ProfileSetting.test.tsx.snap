// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSetting should renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="none"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      [
        {},
        undefined,
      ],
    ]
  }
  testID=""
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    content
  </Text>
</View>
`;

exports[`ProfileSetting should renders with icon 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="none"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      [
        {},
        undefined,
      ],
    ]
  }
  testID=""
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFACD",
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "marginRight": 12,
        },
        {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
        [
          undefined,
          {
            "backgroundColor": "#f5f7fa",
          },
        ],
      ]
    }
  >
    <Text />
  </View>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    content
  </Text>
</View>
`;

exports[`ProfileSetting should renders with link 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="opacity"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      [
        {
          "opacity": undefined,
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFACD",
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "marginRight": 12,
        },
        {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
        [
          undefined,
          {
            "backgroundColor": "#f5f7fa",
          },
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 15,
          },
          [
            {
              "color": "#FFA500",
            },
            {
              "color": "#000000",
            },
          ],
          {
            "fontFamily": "feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    content
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 20,
        },
        undefined,
        {
          "fontFamily": "feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`ProfileSetting should renders with navigation 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  animation="opacity"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      [
        {
          "opacity": undefined,
        },
        undefined,
      ],
    ]
  }
  testID=""
>
  <View
    style={
      [
        {
          "backgroundColor": "#FFFACD",
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "marginRight": 12,
        },
        {
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
        },
        [
          undefined,
          {
            "backgroundColor": "#f5f7fa",
          },
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 15,
          },
          [
            {
              "color": "#FFA500",
            },
            {
              "color": "#000000",
            },
          ],
          {
            "fontFamily": "feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    content
  </Text>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 20,
        },
        undefined,
        {
          "fontFamily": "feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;
